src/Hipace.cpp

```diff
@@ -458,6 +453,17 @@ Hipace::Evolve ()
             }
 
             Wait(step, it);
+            if (it == n_boxes-1) {
+                // Only reset plasma after receiving time step, to use proper density
+                // WARNING: handling of lev is to be improved: this loops over levels, but
+                // lev is set to 0 above.
+                for (int lv=0; lv<=finestLevel(); ++lv) m_multi_plasma.ResetParticles(lv, true);
```

Comment:
@AlexanderSinn @SeverinDiederichs any idea whether the loop over levels could be removed like this?
```suggestion
                # here lev = 0
                m_multi_plasma.ResetParticles(lev, true);
```

Reply:
I don’t think so because level 1 would have to be reset somewhere

Reply:
Thanks, I agree. Could be worth cleaning this up, but that's for another time.